### ISSUE TITLE
`dpi*_release()` and lifetime

### ISSUE THREAD

**Issue by kubo** [rustyhorde/mimir#28]
I know that mimir is in development. If you have already known the following, close this issue.

Could you implement the Drop trait which calls `dpi*_release()`? It should be called at the end of the lifetime of a Rust value associated with an ODPI-C object because ODPI-C manages resources by reference counting. If the reference count in an ODPI-C object isn't decremented by `dpi*_release()` before the end of the lifetime, resources allocated by ODPI-C are not freed until process termination.

Could you remove `add_ref()` and `release()` methods? I could not imagine the situations where these methods are required if the Drop trait is implemented.
In addition, `release()` must be used carefully. If `release()` is called after the reference count reaches zero, it corrupts memory. IMO, methods which may corrupt memory should not be public.

**Comment by CraZySacX** [comment]
Yeah, the only drop I have implemented so far is in Context.  I will keep this issue in place to track the rest.   When I've implemented them, I will remove `add_ref` and `release` from the public API.
